src/uthash.h

```diff
@@ -719,7 +721,7 @@ do {
     (out) = NULL;                                                                \
   }                                                                              \
   while ((out) != NULL) {                                                        \
-    if ((out)->hh.hashv == (hashval) && (out)->hh.keylen == (keylen_in)) {       \
+    if (((out)->hh.hashv == (hashval)) && ((out)->hh.keylen == (keylen_in))) {   \
```

Comment:
This change seems harmless but at the same time I'm not in favor of it because it seems to complicate this expression for no actual gain.

If this is to deal with some MISRA-C rule, could you provide a pointer to some code that uses uthash and needs to compile with MISRA-C rules? Again I feel like the proper workaround might just be to suppress that diagnostic in files that `#include "uthash.h"`.